docker/pytorch/Dockerfile

```diff
@@ -102,11 +108,41 @@ RUN pip install --no-cache-dir --find-links https://download.pytorch.org/whl/tor
 #########
 RUN rm -rf /tmp/*
 
+
+#######################
+# Set the shell to bash
+#######################
+SHELL ["/bin/bash", "-c"]
+
+
+################################
+# Use the correct python version
+################################
+
+# Set the default python by creating our own folder and hacking the path
+# We don't want to use upgrade-alternatives as that will break system packages
+
+ARG COMPOSER_PYTHON_BIN=/composer-python
+
+RUN mkdir -p ${COMPOSER_PYTHON_BIN} && \
+    ln -s $(which python${PYTHON_VERSION}) ${COMPOSER_PYTHON_BIN}/python && \
+    ln -s $(which python${PYTHON_VERSION}) ${COMPOSER_PYTHON_BIN}/python3 && \
+    ln -s $(which python${PYTHON_VERSION}) ${COMPOSER_PYTHON_BIN}/python${PYTHON_VERSION} && \
+    ln -s $(which pip${PYTHON_VERSION}) ${COMPOSER_PYTHON_BIN}/pip && \
+    ln -s $(which pip${PYTHON_VERSION}) ${COMPOSER_PYTHON_BIN}/pip3 && \
+    ln -s $(which pip${PYTHON_VERSION}) ${COMPOSER_PYTHON_BIN}/pip${PYTHON_VERSION} && \
+    # Include this folder, and the local bin folder, on the path 
+    echo "export PATH=~/.local/bin:$COMPOSER_PYTHON_BIN:$PATH" >> /etc/profile && \
+    echo "export PATH=~/.local/bin:$COMPOSER_PYTHON_BIN:$PATH" >> /etc/bash.bashrc && \
+    echo "export PATH=~/.local/bin:$COMPOSER_PYTHON_BIN:$PATH" >> /etc/zshenv
+
+# Ensure that non-interactive shells load /etc/profile
+ENV BASH_ENV=/etc/profile
```

Comment:
Can we use pyenv and avoid doing all the above as well as simplify the python install and version management?  I can set that up if you need a break :-)

Reply:
I totally agree that py-env would be cleaner, but there are a few hesitations I have about it:

1. It builds python from source, rather than using a pre-built python. This will require us to use staged docker builds to avoid polluting the docker image with all of the python build dependencies.
2. py-env seems to be a user-level install. We'd need to make sure that every user gets a properly configured env. These hacks configure it for all users, including root and those without real home directories, etc...

Reply:
1.  It looks like pyenv has different download mirrors but I'm not sure if these are precompile binaries or just source that's being downloaded and compiled but I agree it makes more sense to have a `python` image to stage on if we go down this route, this is originally why we had a separate `python` image to provide this logical separation.
2. Definitely a valid concern, do we expect our users to switch from the `mosaicml` user or add users when working with the docker image?  Regardless being able to set this globally would be ideal so makes sense to keep this as is.

Reply:
Sounds good, I'll leave it as is. I think that it's possible for users to run the image as something other than `mosaicml` (not sure why someone would, but I wouldn't put it past me).